packages/monaco/src/browser/monaco-quick-open-service.ts

```diff
@@ -71,6 +71,13 @@ export class MonacoQuickOpenService extends QuickOpenService {
         this.internalOpen(new MonacoQuickOpenControllerOptsImpl(model, this.keybindingRegistry, options));
     }
 
+    hide(reason?: HideReason): void {
+        const hideReason = reason === HideReason.ELEMENT_SELECTED ? monaco.quickOpen.HideReason.ELEMENT_SELECTED :
```

Comment:
For the sake of readability, please use a switch-case and make sure it will have a return value of `never` in the `default` case. So whenever the monaco enums get updated, we will have a compiler error, instead of silently returning with `CANCELED`. Thanks!

Reply:
> [...] a return value of `never` in the `default` case.

Does that means actually throwing an error in the default case?

Reply:
Could I return `undefined` in `default` case?

```typescript
let hideReason: monaco.quickOpen.HideReason | undefined;
switch (reason) {
  case HideReason.ELEMENT_SELECTED:
    hideReason = monaco.quickOpen.HideReason.ELEMENT_SELECTED;
    break;
 case HideReason.FOCUS_LOST:
    hideReason = monaco.quickOpen.HideReason.FOCUS_LOST;
    break;
  case HideReason.CANCELED:
    hideReason = monaco.quickOpen.HideReason.CANCELED;
    break;
}
this.widget.hide(hideReason);
````

Reply:
OK.

Reply:
Done.